dynamic_array/include/dynamic_array/array.h

```diff
@@ -0,0 +1,305 @@
+#ifndef CPP_DYNAMIC_ARRAY_H_
+#define CPP_DYNAMIC_ARRAY_H_
+
+#include <iostream>
+
+namespace td {
+
+static const int kMinCapacity = 16;
+// https://stackoverflow.com/questions/1100311/what-is-the-ideal-growth-rate-for-a-dynamically-allocated-array
+static const int kGrowthFactor = 2;
+static const int kShrinkFactor = 4;
+
+static const int kIndexNotFound = -1;
+
+template <typename T>
+class Array;
+
+template <typename T>
+void swap(Array<T>& rhs, Array<T>& lhs);
+
+// A dynamic array template
+template <typename T>
+class Array {
+ public:
+  Array();
```

Comment:
Why do we need it?

Reply:
We need to use it. For example, `Array<int> array`.